controllers/topology/reconcile_state.go

```diff
@@ -229,6 +229,79 @@ func (r *ClusterReconciler) reconcileControlPlane(ctx context.Context, s *scope.
 		return errors.Wrapf(err, "failed to update %s", tlog.KObj{Obj: s.Desired.ControlPlane.Object})
 	}
 
+	// If the Control has defined a MachineHealthCheck set the ownerReferences and reconcile it.
+	if s.Desired.ControlPlane.MachineHealthCheck != nil || s.Current.ControlPlane.MachineHealthCheck != nil {
+		var owner *unstructured.Unstructured
+		var err error
+
+		// Get the recently created ControlPlane Object to use as an ownerReference for the MachineHealthCheck.
+		if owner, err = r.getReference(ctx, &corev1.ObjectReference{
+			APIVersion: s.Desired.ControlPlane.Object.GetAPIVersion(),
+			Kind:       s.Desired.ControlPlane.Object.GetKind(),
+			Name:       s.Desired.ControlPlane.Object.GetName(),
+			Namespace:  s.Desired.ControlPlane.Object.GetNamespace(),
+		}); err != nil {
+			return errors.Wrapf(err, "failed to retrieve controlPlane ownerReferences for %s", tlog.KObj{Obj: s.Desired.ControlPlane.MachineHealthCheck})
+		}
+
+		// Set the ControlPlane Object and the Cluster as owners for the MachineHealthCheck
+		s.Desired.ControlPlane.MachineHealthCheck.SetOwnerReferences([]metav1.OwnerReference{
+			*ownerReferenceTo(owner),
+			*ownerReferenceTo(s.Current.Cluster),
+		})
+
+		// Reconcile the current and desired state of the MachineHealthCheck.
+		if err := r.reconcileMachineHealthCheck(ctx, s.Current.ControlPlane.MachineHealthCheck, s.Desired.ControlPlane.MachineHealthCheck); err != nil {
+			return errors.Wrapf(err, "failed to update %s", tlog.KObj{Obj: s.Desired.ControlPlane.MachineHealthCheck})
+		}
+	}
+	return nil
+}
+
+// reconcileMachineHealthCheck creates, updates, deletes or leaves untouched a MachineHealthCheck depending on the difference between the
+// current state and the desired state.
+func (r *ClusterReconciler) reconcileMachineHealthCheck(ctx context.Context, current, desired *clusterv1.MachineHealthCheck) error {
+	log := tlog.LoggerFrom(ctx)
+
+	// If the machineHealthCheck doesn't exist but there is a desired machineHealthCheck attempt to create.
+	if current == nil && desired != nil {
+		log.Infof("Creating %s", tlog.KObj{Obj: desired})
+		if err := r.Client.Create(ctx, desired); err != nil {
+			return errors.Wrapf(err, "failed to create %s", tlog.KObj{Obj: desired})
+		}
+		r.recorder.Eventf(desired, corev1.EventTypeNormal, updateEventReason, "Created %q", tlog.KObj{Obj: desired})
+		return nil
+	}
+
+	// If the machineHealthCheck exists but there is no desired machineHealthCheck attempt to delete.
+	if current != nil && desired == nil {
+		log.Infof("Deleting %s", tlog.KObj{Obj: current})
+		if err := r.Client.Delete(ctx, current); err != nil {
+			return errors.Wrapf(err, "failed to delete %s", tlog.KObj{Obj: current})
+		}
+		r.recorder.Eventf(current, corev1.EventTypeNormal, updateEventReason, "Deleted %q", tlog.KObj{Obj: current})
+		return nil
+	}
+
+	log.Infof("Attempting to Patch %s", tlog.KObj{Obj: desired})
```

Comment:
Let's drop this log line, this is the only place where we are logging "attempting to..."